tests/categorical/test_contingency.py

```diff
@@ -0,0 +1,198 @@
+            [1.7, 1.2, 1.7],
+            [1.7, 2.2, 3.9],
+            [1.6, 1.2, 9.9],
+        ],
+    ],
+    coords=[[10, 20], [0, 1, 2], [5, 6, 7]],
+    dims=["height", "lat", "lon"],
+)
+
+# This truth table shows where the forecast and obs match precisely
+exact_matches = xr.DataArray(
+    [
+        [
+            [True, True, True],
+            [True, False, False],
+            [False, False, False],
+        ],
+        [
+            [False, False, False],
+            [True, False, False],
+            [False, False, False],
+        ],
+    ],
+    coords=[[10, 20], [0, 1, 2], [5, 6, 7]],
+    dims=["height", "lat", "lon"],
+)
+
+# This truth table shows where the forecast and obs match within 0.2
+somewhat_near_matches = xr.DataArray(
+    [
+        [
+            [True, True, True],
+            [True, True, True],
+            [True, True, True],
+        ],
+        [
+            [True, True, True],
+            [True, True, False],
+            [True, False, False],
+        ],
+    ],
+    coords=[[10, 20], [0, 1, 2], [5, 6, 7]],
+    dims=["height", "lat", "lon"],
+)
+
+
+def test_str_view():
+    """
+    Smoke test for string representation
+    """
+    match = scores.categorical.ThresholdEventOperator()
+    table = match.make_table(simple_forecast, simple_obs, event_threshold=1.3)
+    _stringrep = str(table)
+
+
+def test_categorical_table():
+    """
+    Test the basic calculations of the contingency table
+    """
+    match = scores.categorical.ThresholdEventOperator(default_event_threshold=1.3)
+    table = match.make_table(simple_forecast, simple_obs)
+    table2 = match.make_table(simple_forecast, simple_obs, event_threshold=1.3)
+    counts = table.get_counts()
+    actual_table = table.get_table()
+
+    # Test event tables creation matches the stored tables
+    fcst_events, obs_events = match.make_event_tables(simple_forecast, simple_obs)
+    fcst_events2, obs_events2 = match.make_event_tables(simple_forecast, simple_obs, event_threshold=1.3)
+    assert (fcst_events == table.forecast_events).all()
+    assert (fcst_events2 == table.forecast_events).all()
+    assert (obs_events == table.observed_events).all()
+    assert (obs_events2 == table.observed_events).all()
+    assert (table.forecast_events == table2.forecast_events).all()
+    assert (table.observed_events == table2.observed_events).all()
+
+    # Confirm values in the contingency table are correct
+    assert counts["tp_count"] == 9
+    assert counts["tn_count"] == 6
+    assert counts["fp_count"] == 2
+    assert counts["fn_count"] == 1
+    assert counts["total_count"] == 18
+
+    # Confirm that the xarray table object has the correct counts too
+    assert actual_table.sel(contingency="tp_count") == 9
+    assert actual_table.sel(contingency="tn_count") == 6
+    assert actual_table.sel(contingency="fp_count") == 2
+    assert actual_table.sel(contingency="fn_count") == 1
+    assert actual_table.sel(contingency="total_count") == 18
```

Comment:
Optional - you could simplify this to check if `actual_table` equals the expected xarray object in a single line

e.g., assert equal to

```
<xarray.DataArray (contingency: 5)>
array([ 9,  6,  2,  1, 18])
Coordinates:
  * contingency  (contingency) <U11 'tp_count' 'tn_count' ... 'total_count'
 ```

Reply:
I find the current code easier to understand. Feel free to make the change yourself if you do think it's worthwhile however.